client/question/components/LearnerViewDirective.js

```diff
@@ -661,6 +661,63 @@ tie.directive('learnerView', [function() {
         var questionWindowDiv =
             document.getElementsByClassName('tie-question-window')[0];
 
+        var onVisibilityChange = function() {
+          // When a user changes tabs (or comes back), add a SessionPause
+          // or SessionResumeEvent, respectively.
+          var hiddenAttributeName = determineHiddenAttributeNameForBrowser();
+          if (hiddenAttributeName !== null &&
+            document[determineHiddenAttributeNameForBrowser()]) {
+            EventHandlerService.createSessionPauseEvent(
+              SessionIdService.getSessionId());
+          } else {
+            EventHandlerService.createSessionResumeEvent(
+              SessionIdService.getSessionId());
+          }
+        };
+
+        /**
+         * Different browsers call the "hidden" attribute different things.
+         * This method determines what the current browser calls its "hidden"
+         * attribute and returns it.
+         */
+        var determineHiddenAttributeNameForBrowser = function() {
+          if (typeof document.hidden !== 'undefined') {
+            // Opera 12.10 and Firefox 18 and later support
+            return 'hidden';
+          } else if (typeof document.msHidden !== 'undefined') {
+            return 'msHidden';
+          } else if (typeof document.webkitHidden !== 'undefined') {
+            return 'webkitHidden';
+          }
+          return null;
+        };
+
+        var setEventListenerForVisibilityChange = function() {
+          var hiddenAttributeName = determineHiddenAttributeNameForBrowser();
+          var visibilityChange;
```

Comment:
Nit: define this on line 705, so that its scope is limited. The rest of the function does not need to know about it.

Reply:
k